.github/workflows/canary-integration-test.yml

```diff
@@ -16,6 +16,11 @@ defaults:
     # reference: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell
     shell: bash --noprofile --norc -eo pipefail -x {0}
 
+# cancel the in-progress workflow when PR is refreshed.
```

Comment:
What happens in the `push` case where multiple PRs are merged to the same branch (master or release branch)? Those tests should all run to completion. Hopefully only actions in PRs would be cancelled when the PR is updated, but multiple merges to a branch around the same time will not cancel.

Reply:
Yes AFAIU, this is applicable for the pull request update. i experimented on the pull requests.

Reply:
@Madhu-1 can we add condition something like this ```github.event_name != 'pull_request'```? or  this is not required

Reply:
Looks like that's implemented now